c++/test/TestReader.cc

```diff
@@ -137,4 +139,89 @@ namespace orc {
     CheckFileWithSargs("bad_bloom_filter_1.6.11.orc", "ORC C++ 1.6.11");
     CheckFileWithSargs("bad_bloom_filter_1.6.0.orc", "ORC C++");
   }
+
+  /**
+   * Read complextypes_iceberg.orc and verify the resolved selections.
+   *
+   * The ORC file has the following schema:
+   *   struct<
+   *     id:bigint,
+   *     int_array:array<int>,
+   *     int_array_array:array<array<int>>,
+   *     int_map:map<string,int>,
+   *     int_map_array:array<map<string,int>>,
+   *     nested_struct:struct<
+   *       a:int,
+   *       b:array<int>,
+   *       c:struct<
+   *         d:array<array<struct<
+   *           e:int,
+   *           f:string
+   *         >>>
+   *       >,
+   *       g:map<string,struct<
+   *         h:struct<
+   *           i:array<double>
+   *         >
+   *       >>
+   *     >
+   *   >
+   *
+   * @param readIntents TypeReadIntents describing the section.
```

Comment:
typo? "section" => "selection" ?

Reply:
I believe this has been fixed at commit [11d66a2](https://github.com/apache/orc/pull/990/commits/11d66a2035f1d91346e1e44986e7ffd4e20e5b0f)